### PR TITLE
Allow to use selected CiviOffice documents for applications via remote API

### PR DESCRIPTION
**Pull request by dontub**
CiviOffice documents can be associated to funding case types as application template via `FundingApplicationCiviOfficeTemplate` entity. These documents can be rendered via remote API.

systopia-reference: 23716

### CODE REVIEW

File: xml/schema/CRM/Funding/FundingApplicationCiviOfficeTemplate.xml [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    <name>id</name>
+    <autoincrement>true</autoincrement>
+  </primaryKey>
+
+  <field>
+    <!--
+      Without "funding_" so the generated constraint name does not exceed the
+      limit of 64 chars.
+    -->
+    <name>case_type_id</name>
```
Comment (jensschuppe) [inline xml/schema/CRM/Funding/FundingApplicationCiviOfficeTemplate.xml:31]: Oh that's too bad, but `case` without `funding` might be ambiguous with *CiviCase* entities. Will at least an `f` fit in …?

File: Civi/Funding/DocumentRender/CiviOffice/CiviOfficeDocumentRenderer.php [hunk 2]
```diff
@@ -20,33 +20,17 @@
 namespace Civi\Funding\DocumentRender\CiviOffice;
 
 use Civi\Funding\DocumentRender\DocumentRendererInterface;
-use Civi\RemoteTools\Api3\Api3Interface;
-use Webmozart\Assert\Assert;
 
 class CiviOfficeDocumentRenderer implements DocumentRendererInterface {
 
-  private Api3Interface $api3;
+  private CiviOfficeRenderer $renderer;
```
Comment (jensschuppe) [inline Civi/Funding/DocumentRender/CiviOffice/CiviOfficeDocumentRenderer.php:26]: What's the point having this basically forward every method to a new class?

File: ang/crmFunding/afform/afSubmitReload.directive.js [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+'use strict';
+
+/**
+ * Reloads page after successful afform submit.
+ * Usage: <af-form ctrl="afform" funding-af-submit-reload>
+ */
+fundingModule.directive('fundingAfSubmitReload', function() {
+  return {
+    restrict: 'A',
```
Comment (jensschuppe) [inline ang/crmFunding/afform/afSubmitReload.directive.js:25]: Can we have a little in-code documentation on what this means?

**Review by jensschuppe — CHANGES_REQUESTED** [review]

File: xml/schema/CRM/Funding/FundingApplicationCiviOfficeTemplate.xml (hunk 1 above)
Comment (dontub) [reply xml/schema/CRM/Funding/FundingApplicationCiviOfficeTemplate.xml:31]: Actually I'm not sure if `f_case_type_id` (or `fcase_type_id`) would be less confusing... At least to me it looks strange. I'd think it's clear from the context that it has no relation to `CiviCase`.

File: Civi/Funding/DocumentRender/CiviOffice/CiviOfficeDocumentRenderer.php (hunk 2 above)
Comment (dontub) [reply Civi/Funding/DocumentRender/CiviOffice/CiviOfficeDocumentRenderer.php:26]: The `render()` method is not a pure forward. So the behavior of the class didn't change and the new class `CiviOfficeRenderer` can be used in `ApplicationTemplateRenderController`, too.

File: ang/crmFunding/afform/afSubmitReload.directive.js (hunk 3 above)
Comment (dontub) [reply ang/crmFunding/afform/afSubmitReload.directive.js:25]: https://docs.angularjs.org/api/ng/service/$compile#-restrict-

Actually I don't like to copy AngularJS documentation.

**Pull request merged**